flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -875,6 +876,9 @@ public void run() {
 						localTaskOperatorSubtaskStates,
 						asyncDurationMillis);
 
+					if (owner.asynchronousCheckpointExceptionHandler != null) {
```

Comment:
@azagrebin In the normal runtime it will not be null, plus this judgment is mainly for the checkpoint test code (it does not perform the initialization logic for the asynchronousCheckpointExceptionHandler), if not added then the test code will report to the NPE, so in the second commit, I added this judgment.

Reply:
I see, I would rather initialise it in `NoOpStreamTask` to avoid `null` but it would need then more refactoring.

Reply:
I looked at the StreamTask, the initialization logic of the asynchronousCheckpointExceptionHandler is in the invoke method and before the call to the concrete init method. The invoke method has been marked as final. So, I personally feel that we don't have to make this thing too complicated, just to reduce a null judgment. What do you think?

Reply:
ok, I agree that it would need more refactoring to remove this null check.